spec/dynamoid/transaction_write_spec.rb

```diff
@@ -0,0 +1,414 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+# Dynamoid.config.logger.level = :debug
+
+describe Dynamoid::TransactionWrite do
+
+  describe '.execute' do
+    let(:klass) do
+      new_class(class_name: 'Document') do
+        field :name
+      end
+    end
+
+    let(:klass_with_composite_key) do
+      new_class(class_name: 'Cat') do
+        range :age, :integer
+        field :name
+      end
+    end
+
+    let(:klass_with_callback) do
+      new_class(class_name: 'Dog') do
+        field :name
+        field :says
+        field :says_more
+        before_save { self.says = "woof #{self.says_more}" }
+      end
+    end
+
+    let(:klass_with_validation) do
+      new_class do
+        field :name
+        validates :name, length: { minimum: 4 }
+      end
+    end
+
+    before do
+      klass.create_table
+      klass_with_composite_key.create_table
+      klass_with_callback.create_table
+      klass_with_validation.create_table
+    end
+
+    context 'simple primary key' do
+      it 'creates' do
+        obj1 = klass.new(name: "one")
+        obj2 = klass.new
+        obj3 = nil
+        expect(obj1.persisted?).to eql(false)
+        Dynamoid::TransactionWrite.execute do |txn|
+          txn.create! obj1
+          txn.create! obj2, {name: "two"}
```

Comment:
I agree that accepting additional attributes on the model-create is unlikely to be used. Since they will be used on the class-create and it's the same method it will work, however. Were you thinking we should put it extra effort and raise an exception to prevent it from working?

Allowing the extra attributes on all uses of update! does make sense I believe.

Reply:
I believe it's OK as is and checking actual arguments with raising exception or just warning could be a followup.